internal/core/algorithm/ngt/ngt.go

```diff
@@ -156,7 +156,7 @@ const (
 	// -------------------------------------------------------------
 	// dimension constraints
 	// -------------------------------------------------------------.
-	ngtVectorDimensionSizeLimit = 1 << 16
+	ngtVectorDimensionSizeLimit = 1<<32 - 1
```

Comment:
should we expose this variable for external package testing?

Reply:
It would be better :)